flink-state-backends/flink-statebackend-rocksdb/src/test/java/org/apache/flink/contrib/streaming/state/RocksDBOperationsUtilsTest.java

```diff
@@ -78,6 +81,60 @@ public void testPathExceptionOnWindows() throws Exception {
 		}
 	}
 
+	@Test
+	public void testSanityCheckArenaBlockSize() {
+		long testWriteBufferSize = 56 * 1024 * 1024L;
+		long testDefaultArenaSize = testWriteBufferSize / 8;
+		long testValidArenaSize = testWriteBufferSize / 7;
+		long testInvalidArenaSize = testWriteBufferSize / 7 - 8L;
+		List<TestData> tests = Arrays.asList(
+				new TestData(testWriteBufferSize, 0, testInvalidArenaSize, false),
+				new TestData(testWriteBufferSize, testDefaultArenaSize, testInvalidArenaSize, false),
+				new TestData(testWriteBufferSize, 0, testValidArenaSize, true),
+				new TestData(testWriteBufferSize, testDefaultArenaSize, testValidArenaSize, true)
+		);
+
+		for (TestData test : tests) {
+			long writeBufferSize = test.getWriteBufferSize();
+			long arenaBlockSizeConfigured = test.getArenaBlockSizeConfigured();
+			long writeBufferManagerCapacity = test.getWriteBufferManagerCapacity();
+			boolean expected = test.getExpectedResult();
+
+			boolean sanityCheckResult = RocksDBOperationUtils.sanityCheckArenaBlockSize(writeBufferSize, arenaBlockSizeConfigured, writeBufferManagerCapacity);
+			assertThat(sanityCheckResult, is(expected));
+		}
+	}
+
+	private static class TestData {
+		private final long writeBufferSize;
+		private final long arenaBlockSizeConfigured;
+		private final long writeBufferManagerCapacity;
+		private final boolean expectedResult;
+
+		public TestData(long writeBufferSize, long arenaBlockSizeConfigured, long writeBufferManagerCapacity, boolean expectedResult) {
+			this.writeBufferSize = writeBufferSize;
+			this.arenaBlockSizeConfigured = arenaBlockSizeConfigured;
+			this.writeBufferManagerCapacity = writeBufferManagerCapacity;
+			this.expectedResult = expectedResult;
+		}
+
+		public long getWriteBufferSize() {
+			return writeBufferSize;
+		}
+
+		public long getArenaBlockSizeConfigured() {
+			return arenaBlockSizeConfigured;
+		}
+
+		public long getWriteBufferManagerCapacity() {
+			return writeBufferManagerCapacity;
+		}
+
+		public boolean getExpectedResult() {
+			return expectedResult;
+		}
+	}
```

Comment:
```suggestion
	public void testSanityCheckArenaBlockSize() {
		long testWriteBufferSize = 56 * 1024 * 1024L;
		long testDefaultArenaSize = RocksDBMemoryControllerUtils.calculateRocksDBDefaultArenaBlockSize(testWriteBufferSize);
		long testWriteBufferCapacityBoundary = testDefaultArenaSize * 8 / 7;
		assertThat("The sanity check result is incorrect with default arena block size",
			RocksDBOperationUtils.sanityCheckArenaBlockSize(testWriteBufferSize, 0, testWriteBufferCapacityBoundary),
			is(true));
		assertThat("The sanity check should pass when the configured arena block size is small enough.",
			RocksDBOperationUtils.sanityCheckArenaBlockSize(testWriteBufferSize, testDefaultArenaSize - 1, testWriteBufferCapacityBoundary),
			is(true));
		assertThat("The sanity check should fail when the configured arena block size is too big.",
			RocksDBOperationUtils.sanityCheckArenaBlockSize(testWriteBufferSize, testDefaultArenaSize + 1, testWriteBufferCapacityBoundary),
			is(false));
	}
```
Sorry but I'm still not satisfied with this test and suggest to further simplify it and adding some hints if any of the test fails.

And although this test is also implementation bounded, it checks/guards the result of multiple calculations, so I think we should keep it.

We will also need to remove useless imports if the suggestion is accepted.

Reply:
I changed the test `testSanityCheckArenaBlockSize` to the suggested version and added one more `assertThat`. Also, minor tweaks in the error messages in asserts.